src/Arcus.Messaging.Abstractions.ServiceBus/Extensions/ServiceBusMessageHandlerExtensions.cs

```diff
@@ -109,6 +109,7 @@ public static ServiceBusMessageHandlerCollection WithServiceBusMessageHandler<TM
         /// <typeparam name="TMessageHandler">The type of the fallback message handler.</typeparam>
         /// <param name="handlers">The handlers to add the fallback message handler to.</param>
         /// <exception cref="ArgumentNullException">Thrown when the <paramref name="handlers"/> is <c>null</c>.</exception>
+        [Obsolete("Will be removed in v3.0, please use the Azure service bus operations on the " + nameof(AzureServiceBusMessageContext) + " instead of defining fallback message handlers")]
         public static ServiceBusMessageHandlerCollection WithServiceBusFallbackMessageHandler<TMessageHandler>(
```

Comment:
It's not entirely clear to me how the Complete/Abandon/Deadletter methods on the `AzureServiceBusMessageContext` will replace the fallback messagehandlers ?  (Might be because I never ever used the Fallback mechanism).

Reply:
The fallback message handlers were usually used to call Service bus-operations. Like, when you want to have a custom dead-letter message in case none of the message handlers match. To simplify the messaging system, removing the fallback handlers will also make it more easy/flexible for consumers to use the Service bus operations, which having to implement fallback message handlers.

Reply:
Ah, if that was the idea of the fallback handlers, I see :)  But, IMHO we've improved / simplified this a bit by making sure that the MessageRouter (I think) is now deadlettering the message itself if no suitable MessageHandler can be found.

Reply:
Yes, indeed.